synapse/appservice/api.py

```diff
@@ -231,13 +231,26 @@ async def push_bulk(
                 json_body=body,
                 args={"access_token": service.hs_token},
             )
+            logger.debug(
+                "push_bulk to %s succeeded! events=%s",
+                uri,
+                [event.get("event_id") for event in events],
+            )
```

Comment:
```suggestion
            if logger.isEnabledFor(logging.DEBUG):
                logger.debug(
                    "push_bulk to %s succeeded! events=%s",
                    uri,
                    [event.get("event_id") for event in events],
                )
```

Otherwise we'll always generate the list, even if the logging isn't set to debug